.github/workflows/build-binaries.yaml

```diff
@@ -0,0 +1,166 @@
+name: Build, Test and Publish Pyinstaller Binaries
+
+on:
+  workflow_call:
+    inputs:
+      production_release:
+        required: true
+        type: string
+      python_version:
+        required: true
+        type: string
+      release_version:
+        required: false
+        type: string
+
+jobs:
+  build-binaries:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false # Continue with other matrix jobs even if one fails
+      matrix:
+        # macos-14 is the Apple Silicon M1 runner (mac os 14)
+        # macos-13 is the last available Intel Mac runner (mac os 13)
+        # See https://github.com/actions/runner-images?tab=readme-ov-file#available-images
+        os: [ubuntu-22.04, ubuntu-22.04-arm, windows-latest, macos-13, macos-14]
+
+    steps:
+      - name: Set signing condition
+        id: signing
+        run: |
+          # Allow signing on main branch non-PR events (regardless of production_release value)
+          if [[ "${{ github.event_name }}" != "pull_request" && \
+                "${{ github.ref_name }}" == "main" ]]; then
+            echo "allowed=true" >> $GITHUB_OUTPUT
+          else
+            echo "allowed=false" >> $GITHUB_OUTPUT
+          fi
+        shell: bash
+
+      - name: Checkout source code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.release_version != '' && format('v{0}', inputs.release_version) || '' }}
+
+      - name: Configure build environment
+        shell: bash
+        run: |
+          artifacts_dir="${{ github.workspace }}${{ runner.os == 'Windows' && '\dist\artifacts' || '/dist/artifacts' }}"
+          mkdir -p $artifacts_dir
+          package_name_version="${{ inputs.release_version != '' && format('-{0}', inputs.release_version) || '' }}"
+          package_name="puya${package_name_version}-${{ runner.os }}_${{ runner.arch }}"
+          echo "PACKAGE_NAME=`echo $package_name | tr '[:upper:]' '[:lower:]'`" >> $GITHUB_ENV
+          echo "ARTIFACTS_DIR=${artifacts_dir}" >> $GITHUB_ENV
+
+      - name: Docker registry login
```

Comment:
Is this needed? A comment below says to run Python setup after a docker container has been run, but I can't see any between those two steps.

Reply:
Ah nevermind, it's in a separate action